scripts/Publish-HelmLibrary.ps1

```diff
@@ -0,0 +1,91 @@
+<#
+.SYNOPSIS
+Publish Helm library chart to github repo 'Defra/adp-helm-repository'.
+
+.DESCRIPTION
+Package Helm library chart.
+Publish Helm library chart to github repo 'Defra/adp-helm-repository' and updates index.yaml file in that repo.
+
+.PARAMETER HelmLibraryPath
+Helm library folder root path. requires to package helm library into .tgz package.
+
+.PARAMETER ChartVersion
+Current Helm library ChartVersion.
+
+.PARAMETER HelmChartRepoPublic
+Helm chart publich url. Requires to index.yaml file.
+#>
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory)] 
+    [string]$HelmLibraryPath,
+    [Parameter(Mandatory)] 
+    [string]$ChartVersion,
+    [Parameter(Mandatory)] 
+    [string]$HelmChartRepoPublic
+)
+
+Set-StrictMode -Version 3.0
+
+[string]$functionName = $MyInvocation.MyCommand
+[datetime]$startTime = [datetime]::UtcNow
+
+[int]$exitCode = -1
+[bool]$setHostExitCode = (Test-Path -Path ENV:TF_BUILD) -and ($ENV:TF_BUILD -eq "true")
+[bool]$enableDebug = (Test-Path -Path ENV:SYSTEM_DEBUG) -and ($ENV:SYSTEM_DEBUG -eq "true")
+
+Set-Variable -Name ErrorActionPreference -Value Continue -scope global
+Set-Variable -Name InformationPreference -Value Continue -Scope global
+
+if ($enableDebug) {
+    Set-Variable -Name VerbosePreference -Value Continue -Scope global
+    Set-Variable -Name DebugPreference -Value Continue -Scope global
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:HelmLibraryPath=$HelmLibraryPath"
+Write-Debug "${functionName}:ChartVersion=$ChartVersion"
+Write-Debug "${functionName}:HelmChartRepoPublic=$HelmChartRepoPublic"
+
+try {
+    Write-Host "Package Helm library chart"
+    helm package $HelmLibraryPath 
```

Comment:
There are many CLI commands in the script. i think we should use the "Command-Invoke" module or check the LASTERRORCODE for all the commands.

Reply:
Added LASTERRORCODE check

Reply:
There are more CLI's I think use the ps helper module to make it consistent with the other scripts.

Reply:
@ppotturi : fixed